engine/sendmessage.go

```diff
@@ -114,15 +114,21 @@ func (p *Engine) sendMessage(ctx context.Context, msg *message.Message) (*notifi
 		return &notification.MessageStatus{State: notification.MessageStateFailedPerm}, nil
 	}
 
+	meta := alertlog.NotificationMetaData{
+		MessageID: msg.ID,
+	}
+
 	status, err := p.cfg.NotificationSender.Send(ctx, notifMsg)
 	if err != nil {
+		p.cfg.AlertLogStore.MustLog(ctx, msg.AlertID, alertlog.TypeNotificationSent, meta)
```

Comment:
In this case, something went wrong before making the attempt so we wouldn't want to log anything. This also could be a test or verification message with no AlertID.

Any case that should be recorded as a sent-but-retry (temp.) or sent-but-failed (perm.) will indicate in the status but with a `nil` err and get logged below.